From 55659a5e7f7557723d23709ab3cb5c72c9177f7f Mon Sep 17 00:00:00 2001
From: Saahil Tomar <quic_saahtoma@quicinc.com>
Date: Mon, 6 Nov 2023 15:01:54 +0530
Subject: [PATCH 494/500] tty: serial: msm: make tx watermark configurable

This patch makes the tx watermark configurable from
dts and setting it to zero for uart console.
By setting tx watermark to zero, serial driver cpu
utilization is getting reduced.

(cherry picked from commit 417eb09975addd2806fc8ad1b346a00e457a310c)
Change-Id: Ia8eb9cc83ca53689ba57b34a284b67c06763c67c
Signed-off-by: Saahil Tomar <quic_saahtoma@quicinc.com>
---
 drivers/tty/serial/msm_serial.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/msm_serial.c b/drivers/tty/serial/msm_serial.c
index 90953e679e38..7629668d40dd 100644
--- a/drivers/tty/serial/msm_serial.c
+++ b/drivers/tty/serial/msm_serial.c
@@ -1126,6 +1126,8 @@ static int msm_set_baud_rate(struct uart_port *port, unsigned int baud,
 	struct msm_port *msm_port = to_msm_port(port);
 	const struct msm_baud_map *entry;
 	unsigned long flags, rate;
+	struct device *dev = msm_port->uart.dev;
+	u32 tx_watermark = 10;
 
 	flags = *saved_flags;
 	spin_unlock_irqrestore(&port->lock, flags);
@@ -1159,7 +1161,8 @@ static int msm_set_baud_rate(struct uart_port *port, unsigned int baud,
 	msm_write(port, watermark, MSM_UART_RFWR);
 
 	/* set TX watermark */
-	msm_write(port, 10, MSM_UART_TFWR);
+	of_property_read_u32(dev->of_node, "tx-watermark", &tx_watermark);
+	msm_write(port, tx_watermark, MSM_UART_TFWR);
 
 	msm_write(port, MSM_UART_CR_CMD_PROTECTION_EN, MSM_UART_CR);
 	msm_reset(port);
-- 
2.34.1

